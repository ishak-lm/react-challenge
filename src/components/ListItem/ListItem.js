import React from 'react'
import PropTypes from 'prop-types'

import S from './ListItem.styles'

import { ProgressBar } from '..'

const ListItem = ({ data, width, color, right, left }) => (
  <S.Li>
    <ProgressBar width={width} color={color} right={right} left={left} />
    {data.map((e, key) => (
      <S.Span key={key}>{e}</S.Span>
    ))}
  </S.Li>
)

ListItem.propTypes = {
  width: PropTypes.number,
  color: PropTypes.string,
  right: PropTypes.bool,
  left: PropTypes.bool,
  data: PropTypes.array.isRequired,
}

ListItem.defaultProps = {
  width: 100,
  color: 'green',
  right: false,
  left: false,
}

export default ListItem
