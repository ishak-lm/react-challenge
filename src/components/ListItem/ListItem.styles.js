import styled from 'styled-components'

const Li = styled.li`
  height: 25px;
  display: flex;
  justify-content: space-around;
  position: relative;
  align-items: center;
`

const Span = styled.span`
  z-index: 1;
`

export default {
  Li,
  Span,
}
