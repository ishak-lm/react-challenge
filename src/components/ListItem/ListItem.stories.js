import React from 'react'
import { withKnobs, boolean, select, number } from '@storybook/addon-knobs'

import ListItem from './ListItem'

export default {
  title: '03 - ListItem',
}

export const StoryListItem = () => (
  <ListItem
    data={['123', '5432', '23424']}
    width={number('Width', 60, { range: true, min: 0, max: 100 })}
    color={select('Color', ['green', 'red'])}
    right={boolean('Right', true)}
    left={boolean('Left', false)}
  />
)

StoryListItem.story = {
  decorators: [withKnobs],
  parameters: { options: { selectedPanel: 'storybook/knobs/panel' } },
}
