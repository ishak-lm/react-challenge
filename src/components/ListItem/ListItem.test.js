/* global describe it expect */
import React from 'react'
import { mount } from 'enzyme'

import ListItem from './ListItem'

describe('ListItem', () => {
  const wrapper = mount(<ListItem data={['234', '235', '34']} />)

  expect(wrapper).toMatchSnapshot()

  it('should have the correct default props', () => {
    expect(wrapper.props()).toMatchObject({
      width: 100,
      color: 'green',
      right: false,
      left: false,
    })
  })
})
