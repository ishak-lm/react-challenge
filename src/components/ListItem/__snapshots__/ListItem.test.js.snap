// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ListItem
  color="green"
  data={
    Array [
      "234",
      "235",
      "34",
    ]
  }
  left={false}
  right={false}
  width={100}
>
  <styled.li>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "rAcOg",
            "rules": Array [
              "
  height: 25px;
  display: flex;
  justify-content: space-around;
  position: relative;
  align-items: center;
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <li
        className="sc-bdVaJa rAcOg"
      >
        <ProgressBar
          color="green"
          left={false}
          right={false}
          width={100}
        >
          <styled.div
            color="green"
            left={false}
            right={false}
            width={100}
          >
            <StyledComponent
              color="green"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "ekulwv",
                    "rules": Array [
                      "
  position: absolute;
  width: ",
                      [Function],
                      ";
  height: 25px;
  background-color: ",
                      [Function],
                      ";
  right: ",
                      [Function],
                      ";
  left: ",
                      [Function],
                      ";
  z-index: 0;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              left={false}
              right={false}
              width={100}
            >
              <div
                className="sc-htpNat ekulwv"
                color="green"
                width={100}
              />
            </StyledComponent>
          </styled.div>
        </ProgressBar>
        <styled.span
          key="0"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "gpNWDq",
                  "rules": Array [
                    "
  z-index: 1;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH gpNWDq"
            >
              234
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          key="1"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "gpNWDq",
                  "rules": Array [
                    "
  z-index: 1;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH gpNWDq"
            >
              235
            </span>
          </StyledComponent>
        </styled.span>
        <styled.span
          key="2"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "gpNWDq",
                  "rules": Array [
                    "
  z-index: 1;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH gpNWDq"
            >
              34
            </span>
          </StyledComponent>
        </styled.span>
      </li>
    </StyledComponent>
  </styled.li>
</ListItem>
`;
